<a href="https://colab.research.google.com/github/CadijaEmbalo/My-Python-project/blob/main/Pandas_Day_1_cadija_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# Requirement already satisfied means library already installed.

In [ ]:
#create report of student with their mark less than 70 at class seven


In [ ]:
# import pandas library
import pandas as pd

In [ ]:
# check version
pd.__version__

'2.2.2'

# Import Dataset

In [ ]:
df = pd.read_csv("student.csv")
# df represent dataframe.
# create df variable and assing it to dataset

In [ ]:
df = pd.read_excel("student.xlsx")

In [ ]:
df

,id,name,class,mark,gender
0,1,John Deo,Four,75,female
1,2,Max Ruin,Three,85,male
2,3,Arnold,Three,55,male
3,4,Krish Star,Four,60,female
4,5,John Mike,Four,60,female
5,6,Alex John,Four,55,male
6,7,My John Rob,Fifth,78,male
7,8,Asruid,Five,85,male
8,9,Tes Qry,Six,78,NaN
9,10,Big John,Four,55,female


# EDA (Exploratory Data Analysis)

In [ ]:
# first 5 rows
df.head()

,id,name,class,mark,gender
0,1,John Deo,Four,75,female
1,2,Max Ruin,Three,85,male
2,3,Arnold,Three,55,male
3,4,Krish Star,Four,60,female
4,5,John Mike,Four,60,female


In [ ]:
# first 10 rows
df.head(10)

,id,name,class,mark,gender
0,1,John Deo,Four,75,female
1,2,Max Ruin,Three,85,male
2,3,Arnold,Three,55,male
3,4,Krish Star,Four,60,female
4,5,John Mike,Four,60,female
5,6,Alex John,Four,55,male
6,7,My John Rob,Fifth,78,male
7,8,Asruid,Five,85,male
8,9,Tes Qry,Six,78,NaN
9,10,Big John,Four,55,female


In [ ]:
# last 5 rows
df.tail()

,id,name,class,mark,gender
30,31,Marry Toeey,Four,88,male
31,32,Binn Rott,Seven,90,female
32,33,Kenn Rein,Six,96,female
33,34,Gain Toe,Seven,69,male
34,35,Rows Noump,Six,88,female


In [ ]:
# last 10 rows
df.tail(10)

,id,name,class,mark,gender
25,26,Crelea,Seven,79,male
26,27,NaN,Three,81,NaN
27,28,Rojj Base,Seven,86,female
28,29,Tess Played,Seven,55,male
29,30,Reppy Red,Six,79,female
30,31,Marry Toeey,Four,88,male
31,32,Binn Rott,Seven,90,female
32,33,Kenn Rein,Six,96,female
33,34,Gain Toe,Seven,69,male
34,35,Rows Noump,Six,88,female


In [ ]:
# head and tail method will return 5 rows by default

In [ ]:
df.tail(1)

,id,name,class,mark,gender
34,35,Rows Noump,Six,88,female


In [ ]:
df.head(1)

,id,name,class,mark,gender
0,1,John Deo,Four,75,female


In [ ]:
# random data sample
df.sample()

,id,name,class,mark,gender
19,20,Jackly,Nine,65,female


In [ ]:
# 5 random samples
df.sample(5)

,id,name,class,mark,gender
15,16,Gimmy,Four,88,male
21,22,Reggid,Seven,55,female
19,20,Jackly,Nine,65,female
4,5,John Mike,Four,60,female
18,19,Tinny,Nine,18,male


In [ ]:
# number of rows and columns
df.shape
# it doesnt have (). its an attribute
#(rows, columns)

(35, 5)

In [ ]:
# dataset information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   id      35 non-null     int64 
 1   name    34 non-null     object
 2   class   34 non-null     object
 3   mark    35 non-null     int64 
 4   gender  33 non-null     object
dtypes: int64(2), object(3)
memory usage: 1.5+ KB


- look at the number of entries and compare it to Non-null count.
- if they are different, that column has a missing value
- name column has 34. 1 name is missing
- class column has 1 missing data
- gender column has 33. 2 info missing

- object data type is string

In [ ]:
# name of the columns
df.columns

Index(['id', 'name', 'class', 'mark', 'gender'], dtype='object')

In [ ]:
# statistical summary for numeric columns only
df.describe()

,id,mark
count,35.000000,35.000000
mean,18.000000,74.657143
std,10.246951,16.401117
min,1.000000,18.000000
25%,9.500000,62.500000
50%,18.000000,79.000000
75%,26.500000,88.000000
max,35.000000,96.000000


- std = standart deviation
- 25%: 25% of the marks of the student are equal to or less than 62.5
- 50% median. 50% of the marks of the student are equal to or less than 79
- if std is close or greater than mean, there are outliers in your data

In [ ]:
# number of rows
len(df)

35

In [ ]:
# rows * columns
df.size
# 35 * 5 = 175

175

In [ ]:
# datatypes of the columns
df.dtypes

,0
id,int64
name,object
class,object
mark,int64
gender,object


In [ ]:
# Counts missing values per column.
df.isnull().sum()

,0
id,0
name,1
class,1
mark,0
gender,2


In [ ]:
# datatypes of the columns
df.dtypes


NameError: name 'df' is not defined

In [ ]:
# import pandas library
import pandas as pd

In [ ]:
# check version
pd.__version__


'2.2.2'

In [ ]:
df= pd.read_excel("student.xlsx")

In [ ]:
# first 5 rows
df.head()

,id,name,class,mark,gender
0,1,John Deo,Four,75,female
1,2,Max Ruin,Three,85,male
2,3,Arnold,Three,55,male
3,4,Krish Star,Four,60,female
4,5,John Mike,Four,60,female


In [ ]:
# name of the columns
df.columns

Index(['id', 'name', 'class', 'mark', 'gender'], dtype='object')

In [ ]:
# datatypes of the columns
df.dtypes


,0
id,int64
name,object
class,object
mark,int64
gender,object


In [ ]:
# Counts missing values per column.
df.isnull().sum()



,0
id,0
name,1
class,1
mark,0
gender,2


In [ ]:
# Counts duplicate rows.

df.duplicated().sum()



np.int64(0)

In [ ]:
df["name"]
#slicing method


,name
0,John Deo
1,Max Ruin
2,Arnold
3,Krish Star
4,John Mike
5,Alex John
6,My John Rob
7,Asruid
8,Tes Qry
9,Big John


In [ ]:
df.gender.tail()

,gender
30,male
31,female
32,female
33,male
34,female


In [ ]:
df.head()

,id,name,class,mark,gender
0,1,John Deo,Four,75,female
1,2,Max Ruin,Three,85,male
2,3,Arnold,Three,55,male
3,4,Krish Star,Four,60,female
4,5,John Mike,Four,60,female


In [ ]:
df.loc[0,4, "name","id","gender"]]

SyntaxError: unmatched ']' (232435727.py, line 1)

In [ ]:
df.loc[0:4 , ["id", "name", "gender"]]
# these are list of the column names ["id", "name", "gender"]


,id,name,gender
0,1,John Deo,female
1,2,Max Ruin,male
2,3,Arnold,male
3,4,Krish Star,female
4,5,John Mike,female


In [ ]:
df.iloc[0,1]

'John Deo'

In [ ]:
df.loc[11:16,["name", "mark"]]

,name,mark
11,Recky,94
12,Kty,88
13,Bigy,88
14,Tade Row,88
15,Gimmy,88
16,Tumyu,54


In [ ]:
#create report of student with their mark less than 70 at class seven


In [ ]:



df[ (df["mark"]<=70) & (df["class"]=="Seven")]

,id,name,class,mark,gender
21,22,Reggid,Seven,55,female
28,29,Tess Played,Seven,55,male
33,34,Gain Toe,Seven,69,male


In [ ]:
#Average mark across all students
df["mark"]

,mark
0,75
1,85
2,55
3,60
4,60
5,55
6,78
7,85
8,78
9,55


In [ ]:
#average mark across female students
df["gender"] == "Female"



,gender
0,False
1,False
2,False
3,False
4,False
5,False
6,False
7,False
8,False
9,False


In [ ]:
#Average mark across male


In [ ]:
#average mark across female students
df["gender"] == "Male"



,gender
0,False
1,False
2,False
3,False
4,False
5,False
6,False
7,False
8,False
9,False


In [ ]:
# Max and min mark
df.mark.max()

96

In [ ]:
#Max mark for male and max mark for female students


In [ ]:
#Average mark for class 'Five'
df[df["class"]=="Five"]

,id,name,class,mark,gender
7,8,Asruid,Five,85,male
17,18,Honny,Five,75,male


In [ ]:
#Average mark for class 'Six' and male students only
df[df["class"]=="six", "male"]


InvalidIndexError: (0     False
1     False
2     False
3     False
4     False
5     False
6     False
7     False
8     False
9     False
10    False
11    False
12    False
13    False
14    False
15    False
16    False
17    False
18    False
19    False
20    False
21    False
22    False
23    False
24    False
25    False
26    False
27    False
28    False
29    False
30    False
31    False
32    False
33    False
34    False
Name: class, dtype: bool, 'male')

In [ ]:
df.loc[(df["gender"]=="male") & (df["mark"]>70)].sort_values(by="class")

,id,name,class,mark,gender
22,23,Herod,Eight,79,male
6,7,My John Rob,Fifth,78,male
7,8,Asruid,Five,85,male
17,18,Honny,Five,75,male
15,16,Gimmy,Four,88,male
30,31,Marry Toeey,Four,88,male
23,24,Tiddy Now,Seven,78,male
24,25,Giff Tow,Seven,88,male
25,26,Crelea,Seven,79,male
1,2,Max Ruin,Three,85,male


In [ ]:
df.sort_values(by="mark").head()
df.sort_values(by="name").head()



In [ ]:
df.loc[(df["gender"]=="male") & (df["mark"]>70)].sort_values(by="class")

,id,name,class,mark,gender
22,23,Herod,Eight,79,male
6,7,My John Rob,Fifth,78,male
7,8,Asruid,Five,85,male
17,18,Honny,Five,75,male
15,16,Gimmy,Four,88,male
30,31,Marry Toeey,Four,88,male
23,24,Tiddy Now,Seven,78,male
24,25,Giff Tow,Seven,88,male
25,26,Crelea,Seven,79,male
1,2,Max Ruin,Three,85,male


In [ ]:
# female students who are above the classroom average. Sort the results in descending order.



In [ ]:
df.loc[(df["gender"]=="female) & (df["mark"]>70)].sort_values(by="results")

SyntaxError: unterminated string literal (detected at line 1) (4021606071.py, line 1)

In [ ]:
df.loc[(df["gender"] =="female") & (df["mark"]>df.mark.mean())].sort_values( by="mark", ascending=False)

,id,name,class,mark,gender
32,33,Kenn Rein,Six,96,female
11,12,Recky,Six,94,female
31,32,Binn Rott,Seven,90,female
10,11,Ronald,Six,89,female
34,35,Rows Noump,Six,88,female
12,13,Kty,Seven,88,female
13,14,Bigy,Seven,88,female
27,28,Rojj Base,Seven,86,female
29,30,Reppy Red,Six,79,female
0,1,John Deo,Four,75,female


In [ ]:
df["name"].value_counts()

,count
name,
John Deo,1
Max Ruin,1
Arnold,1
Krish Star,1
John Mike,1
Alex John,1
My John Rob,1
Asruid,1
Tes Qry,1


In [ ]:
df['gender'].value_counts()

,count
gender,
male,17
female,16


In [ ]:

df['mark'].value_counts()

,count
mark,
88,7
55,5
78,3
79,3
75,2
60,2
85,2
69,2
89,1


In [ ]:

df['class'].value_counts()

,count
class,
Seven,10
Four,8
Six,7
Three,3
Nine,2
Five,2
Fifth,1
Eight,1


In [ ]:

df['name'].value_counts()

,count
name,
John Deo,1
Max Ruin,1
Arnold,1
Krish Star,1
John Mike,1
Alex John,1
My John Rob,1
Asruid,1
Tes Qry,1


In [ ]:
df['mark'].value_counts()

,count
mark,
88,7
55,5
78,3
79,3
75,2
60,2
85,2
69,2
89,1


In [ ]:
#access the third row and the second column in a DataFrame using .iloc
df.iloc[1,2]


'Three'

In [ ]:
df.loc[2,1]

KeyError: 1

In [ ]:
df.iloc[2,1]

'Arnold'

In [ ]:
df.loc[1,2]

KeyError: 2